services/managers/market_manager.py

```diff
@@ -0,0 +1,145 @@
+    SQL_CHECK_EMAIL = r"SELECT email FROM fos_user WHERE email = %s"
+    SQL_CHECK_USERNAME = r"SELECT username FROM fos_user WHERE username = %s"
+    SQL_UPDATE_USER = r"UPDATE fos_user SET password = %s, salt = %s, enabled = '1' WHERE username = %s"
+
+
+    @staticmethod
+    def __santatize_username(username):
+        sanatized = username.replace(" ", "_")
+        sanatized = sanatized.replace("'", "")
+        return sanatized.lower()
+
+    @staticmethod
+    def __generate_random_pass():
+        return os.urandom(8).encode('hex')
+
+    @staticmethod
+    def check_username(username):
+        logger.debug("Checking alliance market username %s" % username)
+        cursor = connections['market'].cursor()
+        cursor.execute(marketManager.SQL_CHECK_USERNAME, [marketManager.__santatize_username(username)])
+        row = cursor.fetchone()
+        if row:
+            logger.debug("Found user %s on alliance market" % username)
+            return True
+        logger.debug("User %s not found on alliance market" % username)
+        return False
+
+    @staticmethod
+    def check_user_email(username, email):
+        logger.debug("Checking if alliance market email exists for user %s" % username)
+        cursor = connections['market'].cursor()
+        cursor.execute(marketManager.SQL_CHECK_EMAIL, [email])
+        row = cursor.fetchone()
+        if row:
+            logger.debug("Found user %s email address on alliance market" % username)
+            return True
+        logger.debug("User %s email address not found on alliance market" % username)
+        return False
+
+
+    @staticmethod
+    def add_user(username, email, characterid, charactername):
+        logger.debug("Adding new market user %s" % username)
+        plain_password = marketManager.__generate_random_pass()
+        hash = bcrypt.encrypt(plain_password, rounds=13)
+        hash_result = hash
+        rounds_striped = hash_result.strip('$2a$13$')
+        salt = rounds_striped[:22]
+        username_clean = marketManager.__santatize_username(username)
+        if marketManager.check_username(username)== False:
+            if marketManager.check_user_email(username, email) == False:
+                try:
+                    logger.debug("Adding user %s to alliance market" % username)
+                    cursor = connections['market'].cursor()
+                    cursor.execute(marketManager.SQL_ADD_USER, [username_clean, username_clean, email, email, salt,
+                                                                hash, characterid, charactername])
+                    return username_clean, plain_password
+                except:
+                    logger.debug("Unsuccessful attempt to add market user %s" % username)
+                    return None
+            else:
+                logger.debug("Alliance market email %s already exists Updating instead" % email)

```

Comment:
why this check? as far as I know email addresses aren't unique, this could break market accounts

[User model](https://docs.djangoproject.com/en/1.9/ref/contrib/auth/#fields)